pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -169,8 +179,55 @@ func New(options Options) Index {
 		return model.ConfigKey{Kind: kind.AuthorizationPolicy, Name: i.Authorization.Name, Namespace: i.Authorization.Namespace}
 	}), false)
 
+	statusQueue := NewStatusQueue()
+
+	serviceEntriesWriter := kclient.NewWriteClient[*networkingclient.ServiceEntry](options.Client)
+	servicesWriter := kclient.NewWriteClient[*v1.Service](options.Client)
 	// these are workloadapi-style services combined from kube services and service entries
 	WorkloadServices := a.ServicesCollection(Services, ServiceEntries, Waypoints, Namespaces)
+	WorkloadServices.Register(func(o krt.Event[model.ServiceInfo]) {
+		if o.Event == controllers.EventDelete {
+			return
+		}
+		obj := o.Latest()
+		client := kclient.ToPatcher(servicesWriter)
+		if obj.Source == kind.ServiceEntry {
+			client = kclient.ToPatcher(serviceEntriesWriter)
+		}
+		statusQueue.Push(func() error {
+			var cond *v1alpha1.IstioCondition
+			if obj.Waypoint.ResourceName != "" {
+				cond = &v1alpha1.IstioCondition{
+					Type:               "istio.io/WaypointBound",
+					Status:             string(metav1.ConditionTrue),
+					LastProbeTime:      nil,
+					LastTransitionTime: timestamppb.Now(),
+					Reason:             "WaypointAccepted",
+					Message:            fmt.Sprintf("Successfull attached to waypoint %v", obj.Waypoint.ResourceName),
+				}
+			} else if obj.Waypoint.Error != nil {
+				cond = &v1alpha1.IstioCondition{
+					Type:               "istio.io/WaypointBound",
+					Status:             string(metav1.ConditionFalse),
+					LastProbeTime:      nil,
+					LastTransitionTime: timestamppb.Now(),
+					Reason:             obj.Waypoint.Error.Reason,
+					Message:            obj.Waypoint.Error.Message,
+				}
+			}
+			// TODO: how can we remove?
+			if cond != nil {
+				condj, err := json.Marshal(cond)
+				if err != nil {
+					return err
+				}
+				patch := fmt.Sprintf(`{"status":{"conditions":[%s]}}`, condj)
+				return client.PatchStatus(obj.Name, obj.Namespace, types.MergePatchType, []byte(patch))
+			}
+			return nil
```

Comment:
add a `wantsWaypoint bool` to the waypoint struct to denote if this wants a waypoint at all? if no error, no name and no wants then remove the bound condition?